.github/workflows/main.yml

```diff
@@ -15,37 +15,44 @@ permissions:
 
 jobs:
   test:
-    name: Test on OTP ${{matrix.otp}} + Elixir ${{matrix.elixir}}
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        # TODO: Consider which matrix combinations are worth testing
-        # otp: ['25', '26', '27']
-        # elixir: ['1.14', '1.15', '1.16', '1.17', '1.18']
-        otp: ['27']
-        elixir: ['1.18']
+        include:
+          - pair:
+              otp: "28"
+              elixir: "1.18"
+            lint: true
+          - pair:
+              otp: "24"
+              elixir: "1.12"
```

Comment:
Use the minimum OTP (24) supported by `setup/beam`, see https://github.com/erlef/setup-beam?tab=readme-ov-file#compatibility-between-operating-system-and-erlangotp

And the minimum Elixir (1.12) support by OTP 24, see https://hexdocs.pm/elixir/1.18.4/compatibility-and-deprecations.html#between-elixir-and-erlang-otp

Reply:
OTP is long dead, let's use the currently supported 27 and 28.

How come the approach to matrix testing has changed here?

Reply:
Testing only the minimal and maximum version of Elixir and Erlang.